packages/react-link/src/components/Link/useLinkState.ts

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { getCode, EnterKey, SpacebarKey } from '@fluentui/keyboard-key';
+import { LinkState } from './Link.types';
+
+/**
+ * The useLink hook processes the Link draft state.
+ * @param state - Link draft state to mutate.
+ */
+export const useLinkState = (state: LinkState) => {
+  const { as, disabled, disabledFocusable, href, onClick, onKeyDown: onKeyDownCallback, type } = state;
+
+  state.onClick = (ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>) => {
+    if (disabled) {
+      ev.preventDefault();
+    } else {
+      onClick?.(ev);
+    }
+  };
+
+  const onKeyDown = (ev: React.KeyboardEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>) => {
+    const keyCode = getCode(ev);
+    const isEnterOrSpaceKey = keyCode === EnterKey || keyCode === SpacebarKey;
+    if (disabled && isEnterOrSpaceKey) {
+      ev.preventDefault();
+      ev.stopPropagation();
+    } else {
+      onKeyDownCallback?.(ev);
+
+      if (onClick && isEnterOrSpaceKey) {
```

Comment:
```suggestion
      if (onClick && isEnterOrSpaceKey && ['a', 'button'].indexOf(as) === -1) {
```

Reply:
This check is not needed here as I'm just setting the `onKeyDown` function when `as` is not `a` or `button` in lines 52 and 58 below.

Reply:
Oh I see, thx for pointing that out